src/sqlfluff/dialects/dialect_redshift.py

```diff
@@ -1816,6 +1816,123 @@ class ShowDatasharesStatementSegment(BaseSegment):
     )
 
 
+class CreateRlsPolicyStatementSegment(BaseSegment):
+    """A `CREATE RLS POLICY` statement.
+
+    https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_RLS_POLICY.html
+    """
+
+    type = "create_rls_policy_statement"
+    match_grammar = Sequence(
+        "CREATE",
+        "RLS",
+        "POLICY",
+        Ref("ObjectReferenceSegment"),
+        Sequence(
+            Ref.keyword("WITH"),
+            Bracketed(
+                Delimited(
+                    Sequence(
+                        Ref("ColumnReferenceSegment"),
+                        Ref("DatatypeSegment"),
+                    ),
+                ),
+            ),
+            Sequence(
+                Ref.keyword("AS", optional=True),
+                Ref("AliasExpressionSegment"),
+                optional=True,
+            ),
+            optional=True,
+        ),
+        Sequence(
+            "USING",
+            Bracketed(Ref("ExpressionSegment")),
+        ),
+    )
+
+
+class AttachRlsPolicyStatementSegment(BaseSegment):
+    """An `ATTACH RLS POLICY` statement.
+
+    https://docs.aws.amazon.com/redshift/latest/dg/r_ATTACH_RLS_POLICY.html
+    """
+
+    type = "attach_rls_policy_statement"
+    match_grammar = Sequence(
+        "ATTACH",
+        "RLS",
+        "POLICY",
+        Ref("ObjectReferenceSegment"),
+        "ON",
+        Ref.keyword("TABLE", optional=True),
+        Delimited(
+            Ref("TableReferenceSegment"),
+        ),
+        "TO",
+        Delimited(
+            OneOf(
+                Sequence(
+                    Ref.keyword("ROLE", optional=True),
+                    Ref("RoleReferenceSegment"),
+                ),
+                "PUBLIC",
+            ),
+        ),
+    )
+
+
+class DetachRlsPolicyStatementSegment(BaseSegment):
+    """A `DETACH RLS POLICY` statement.
+
+    https://docs.aws.amazon.com/redshift/latest/dg/r_DETACH_RLS_POLICY.html
+    """
+
+    type = "detach_rls_policy_statement"
+    match_grammar = Sequence(
+        "DETACH",
+        "RLS",
+        "POLICY",
+        Ref("ObjectReferenceSegment"),
+        "ON",
+        Ref.keyword("TABLE", optional=True),
+        Delimited(
+            Ref("TableReferenceSegment"),
+        ),
+        "FROM",
+        Delimited(
+            OneOf(
+                Sequence(
+                    Ref.keyword("ROLE", optional=True),
+                    Ref("RoleReferenceSegment"),
+                ),
+                "PUBLIC",
+            ),
+        ),
+    )
```

Comment:
I don't really like that this is basically a duplicate of `AttachRlsPolicyStatementSegment` above. I can think of a couple of solutions:
- Merge `AttachRlsPolicyStatementSegment` and `DetachRlsPolicyStatementSegment` into a single segment, but I haven't found examples of a similar approach in the codebase. Is this not recommended?
- Copy `AttachRlsPolicyStatementSegment` match_grammar in `DetachRlsPolicyStatementSegment` and replacing the keyword. Same as above, I haven't found examples of this anywhere else in the dialect code.

Let me know what you think is the correct approach and I'll happily edit that :relaxed:

Reply:
I like just having a single segment that does both. Maybe like `ManageRlsPolicyStatementSegment` or whatever name makes sense?

Then it can start like:

```
match_grammar = Sequence(
        OneOf("ATTACH", "DETACH"),
        "RLS",
        "POLICY",
```

Reply:
Good shout with the name, I was struggling to come up with something; PR is updated :+1: